src/main/java/com/google/sps/servlets/MatchServlet.java

```diff
@@ -28,54 +29,36 @@
   @Override
   public void init(ServletConfig config) throws ServletException {
     super.init(config);
-
-    //won't need ServletContext once we switch to PersistentMatchRepository
-    MatchRepository matchRepository = new NonPersistentMatchRepository();
-    getServletContext().setAttribute("matchRepository", matchRepository);
   }
 
-  public User getLoggedInUser() {
-    User currentUser;
-    NonPersistentUserRepository userRepository = new NonPersistentUserRepository();
-    
-    UserService userService = UserServiceFactory.getUserService();
-    com.google.appengine.api.users.User currentGoogleUser = userService.getCurrentUser();
-    if(currentGoogleUser == null) {
-      currentUser = null;
-    } else {
-      currentUser = userRepository.getUser(currentGoogleUser);
-    }
-
-    return currentUser;
-  }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
     Gson gson = new Gson();
     response.setContentType("application/json;");
-    ServletContext servletContext = getServletContext();
-    User currentUser = getLoggedInUser();
+
+    PersistentUserRepository userRepository = new PersistentUserRepository().getInstance();
+    SessionContext sessionContext = new SessionContext(userRepository);
 
     //is user logged in?
-    if(currentUser == null) {
+    if(!sessionContext.isUserLoggedIn()) {
         //return null for matches so that page redirects to logged out homepage
         response.getWriter().println(gson.toJson(null));
     } else {
-        MatchRepository matchRepository = (MatchRepository) servletContext.getAttribute("matchRepository");
-        Collection<User> matches = matchRepository.getMatchesForUser(currentUser);
-
+        PersistentMatchRepository matchRepository = new PersistentMatchRepository().getInstance();
+        Collection<User> matches = matchRepository.getMatchesForUser(sessionContext.getLoggedInUser());
         response.getWriter().println(gson.toJson(matches));
     }
   }
 
   @Override
   public synchronized void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException {
-    ServletContext servletContext = getServletContext();
-    MatchRepository matchRepository = (MatchRepository) servletContext.getAttribute("matchRepository");
-    User currentUser = getLoggedInUser();
+    PersistentMatchRepository matchRepository = new PersistentMatchRepository().getInstance();
+    PersistentUserRepository userRepository = new PersistentUserRepository().getInstance();
+    SessionContext sessionContext = new SessionContext(userRepository);
 
-    if (currentUser == null) {
+    if (!sessionContext.isUserLoggedIn()) {
         throw new IOException("Logged out user cannot access POST request.");
```

Comment:
If you use a command line tool (e.g. curl), you can send a POST request to the servlet without logging in:
```
curl -X POST https://techish.uc.r.appspot.com/matches
```

Reply:
I changed this to just print an error.